compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -3351,3 +3400,22 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         }
     }
 }
+
+fn deny_non_region_late_bound(
```

Comment:
Should this be checked as part of feature gates?

Reply:
Well, I'm doing this during astconv because we still need to insert an `tcx.ty_error()` for these types. I'm trying to avoid having to teach the type relation code how to relate late-bound ty/ct -- we don't really need that logic anywhere else except for non-well-formed `for<T> fn(T)`, for example.